packages/workspace/src/workspace/nacl_files/nacl_files_source.ts

```diff
@@ -376,4 +388,8 @@ export const naclFilesSource = (
   naclFilesStore: DirectoryStore,
   cache: ParseResultCache,
   staticFileSource: StaticFilesSource,
-): NaclFilesSource => buildNaclFilesSource(naclFilesStore, cache, staticFileSource)
+  parsedFiles?: ParsedNaclFile[],
+): NaclFilesSource => {
+  const state = parsedFiles ? buildNaclFilesState(parsedFiles, {}) : undefined
```

Comment:
We should avoid implicit comparisons to `undefined`, we had too many bugs because of it
```suggestion
  const state = parsedFiles !== undefined ? buildNaclFilesState(parsedFiles, {}) : undefined
```